turtleFSI/problems/__init__.py

```diff
@@ -222,21 +220,40 @@ def save_files_visualization(visualization_folder, dvp_, t, save_deg, mesh, **na
     v = dvp_["n"].sub(1, deepcopy=True)
     p = dvp_["n"].sub(2, deepcopy=True)
 
-    # New functions mimicing higher-order visualization fies
-    if save_deg > 1:
-        d = namespace["save_deg_interpolator"](d, namespace["FSdv_viz"])
-        v = namespace["save_deg_interpolator"](v, namespace["FSdv_viz"])
-        p = namespace["save_deg_interpolator"](p, namespace["FSp_viz"])
-
-    # Name function
-    d.rename("Displacement", "d")
-    v.rename("Velocity", "v")
-    p.rename("Pressure", "p")
-
-    # Write results
-    namespace["d_file"].write(d, t)
-    namespace["v_file"].write(v, t)
-    namespace["p_file"].write(p, t)
+    if save_deg > 1: # To save higher-order nodes:
+
+        # Create lower-order function for visualization on refined mesh
+        d_viz = Function(namespace["FSdv_viz"])
+        v_viz = Function(namespace["FSdv_viz"])
+        p_viz = Function(namespace["FSp_viz"])
+
+        # Interpolate by creating a transfer matrix between higher degree and lower degree visualization function spaces
+        dv_trans = PETScDMCollection.create_transfer_matrix(namespace["FSdv"],namespace["FSdv_viz"])
+        p_trans = PETScDMCollection.create_transfer_matrix(namespace["FSp"],namespace["FSp_viz"])
```

Comment:
This looks a bit ineffective. I would have initiated `d_viz`, `v_viz`, `p_viz", `dv_trans`, `p_trans` previously in the function, like was done with the fenicstools solution, or is the transver_matrix dependent on the mesh movement? I would still have saved the functions.